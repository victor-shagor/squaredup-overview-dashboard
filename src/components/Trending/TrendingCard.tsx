import React from "react";
import styled from "styled-components";
import { TrendingCardProps } from "constants/interfaces";

const TrendingCard = ({ className, loading, array }: TrendingCardProps) => {
  return (
    <Styling className={className && className}>
      {loading && <div data-testid="loading-state">Loading...</div>}
      {!array.length ? (
        <p data-testid="empty-state">No data</p>
      ) : (
        array.map((role: { _id: string; count: number }, index: number) => (
          <div key={index} className="container">
            <div className="nameDiv">
              <div className="BulletDiv"></div>
              <div className="role">
                <p>{role._id}</p>
              </div>
            </div>
            <div className="jobs">{role.count} jobs</div>
          </div>
        ))
      )}
    </Styling>
  );
};
export default TrendingCard;

const Styling = styled.div`
  background-color: #ffffff;
  width: 48%;
  min-height: 300px;
  .container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 60px;
    border-bottom: 1px solid #f1f1f3;
    cursor: pointer;
  }
  .nameDiv {
    display: flex;
  }
  .namediv {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-bottom: 5px;
    height: 40px;
    width: 40px;
    border-radius: 50%;
    margin-right: 10px;
  }
  .role {
    margin-top: 10px;
    font-size: 12px;
  }
  .BulletDiv {
    background-color: #1264f7;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-top: 14px;
    height: 5px;
    width: 5px;
    border-radius: 50%;
    margin-right: 10px;
    font-size: 14px;
    color: #1264f7;
  }
  .more {
    height: 40px;
    color: #0a5ff7;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  .jobs {
    font-size: 12px;
  }
  .trending {
    display: flex;
    justify-content: space-between;
    width: 100%;
  }
  @media screen and (max-width: 900px) {
    width: 80%;
  }
  /* @media screen and (max-width: 900px) and (min-width: 700px) {
    .sidecard {
      display: none;
    }
  } */
`;
