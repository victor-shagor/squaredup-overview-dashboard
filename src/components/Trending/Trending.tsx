import { TrendingCardProps } from "constants/interfaces";
import { Card } from "pages/overview/overview.styles";
import TrendingCard from "./TrendingCard";

const Trending = ({ array, loading }: TrendingCardProps) => {
  return (
    <Card
      height={"25rem"}
      width={"46.4%"}
      className="trends"
      data-testid="trending-container"
    >
      <div className="details">
        <p>Trending roles</p>
      </div>
      <div>
        <div className="trending">
          <TrendingCard
            array={array.slice(0, array.length / 2)}
            loading={loading}
          />
          <TrendingCard
            loading={loading}
            array={array.slice(array.length / 2, array.length)}
            className={array.length > 5 ? "sidecard" : "sidecard hide"}
          />
        </div>
        <div className="more"></div>
      </div>
    </Card>
  );
};

export default Trending;
