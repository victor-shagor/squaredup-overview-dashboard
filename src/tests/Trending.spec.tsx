import { render, screen } from "@testing-library/react";
import Trending from "components/Trending/Trending";
import TrendingCard from "components/Trending/TrendingCard";
import renderer from "react-test-renderer";

jest.mock("react-apexcharts", () => {
  return {
    __esModule: true,
    default: () => {
      return <div />;
    },
  };
});

const mockData = [
  {
    _id: "Engineer",
    count: 6,
  },
  {
    _id: "Nursing",
    count: 5,
  },
  {
    _id: "Doctor",
    count: 5,
  },
];

describe("Trending components", () => {
  it("should render component with the correct style", () => {
    render(<Trending array={mockData} />);
    expect(screen.getByTestId("trending-container")).toHaveStyle(
      `height:25rem`
    );
    expect(screen.getByTestId("trending-container")).toHaveStyle(`width:46.4%`);
  });

  it("should render no data if array is empty", () => {
    render(<TrendingCard array={[]} />);
    expect(screen.getByTestId("empty-state")).toBeInTheDocument();
  });

  it("should render loading state if loading is true", () => {
    render(<TrendingCard array={[]} loading={true} />);
    expect(screen.getByTestId("loading-state")).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Trending array={mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
