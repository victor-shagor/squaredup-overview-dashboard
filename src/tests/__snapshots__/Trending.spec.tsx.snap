// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gsnTZi kBCRrN trends"
  data-testid="trending-container"
  height="25rem"
  width="46.4%"
>
  <div
    className="details"
  >
    <p>
      Trending roles
    </p>
  </div>
  <div>
    <div
      className="trending"
    >
      <div
        className="sc-dkzDqf bfkrXT"
      >
        <div
          className="container"
        >
          <div
            className="nameDiv"
          >
            <div
              className="BulletDiv"
            />
            <div
              className="role"
            >
              <p>
                Engineer
              </p>
            </div>
          </div>
          <div
            className="jobs"
          >
            6
             jobs
          </div>
        </div>
      </div>
      <div
        className="sc-dkzDqf bfkrXT sidecard hide"
      >
        <div
          className="container"
        >
          <div
            className="nameDiv"
          >
            <div
              className="BulletDiv"
            />
            <div
              className="role"
            >
              <p>
                Nursing
              </p>
            </div>
          </div>
          <div
            className="jobs"
          >
            5
             jobs
          </div>
        </div>
        <div
          className="container"
        >
          <div
            className="nameDiv"
          >
            <div
              className="BulletDiv"
            />
            <div
              className="role"
            >
              <p>
                Doctor
              </p>
            </div>
          </div>
          <div
            className="jobs"
          >
            5
             jobs
          </div>
        </div>
      </div>
    </div>
    <div
      className="more"
    />
  </div>
</div>
`;
